client/client.go

```diff
@@ -1384,6 +1385,9 @@ func (c *Client) setupNode() error {
 	node := c.config.Node
 	if node == nil {
 		node = &structs.Node{}
+		if c.config.DefaultIneligible {
+			node.SchedulingEligibility = api.NodeSchedulingIneligible
+		}
```

Comment:
This nil check is a safety check to avoid a panic in the event the node config object is not correctly set. If we chase the function calls from this [`client.setupNode`](https://github.com/hashicorp/nomad/blob/24eb703e740ba5e2859312979e7b23e871507781/client/client.go#L420) call we see:

- it is called by `client.NewClient` which comes from [`agent.setupClient`](https://github.com/hashicorp/nomad/blob/24eb703e740ba5e2859312979e7b23e871507781/command/agent/agent.go#L924)
- the `agent.setupClient` function includes a call to [`agent.clientConfig`](https://github.com/hashicorp/nomad/blob/24eb703e740ba5e2859312979e7b23e871507781/command/agent/agent.go#L489)
- `agent.clientConfig` calls [`convertClientConfig`](https://github.com/hashicorp/nomad/blob/24eb703e740ba5e2859312979e7b23e871507781/command/agent/agent.go#L555) which performs a number of [node struct setup](https://github.com/hashicorp/nomad/blob/24eb703e740ba5e2859312979e7b23e871507781/command/agent/agent.go#L617-L628) items

Therefore during normal and hopefully every client setup this condition is never hit and therefore the eligibility toggle is not used.